src/components/EthereumProviderDetector.tsx

```diff
@@ -11,6 +12,9 @@ export default function EthereumProviderDetector() {
 
         async function detect() {
             try {
+                if (!window.ethereum) {
+                    throw new Error('Ethereum provider could not be detected.')
+                }
                 const provider =
                     (await detectEthereumProvider()) as MetaMaskInpageProvider
```

Comment:
Is the separate check for `ethereum` above necessary? That `detect` utility should be able to take care of the missing `window.ethereum`. It actually does a couple of other things on top of that.

Reply:
It does detect it but pipes the error weirdly! Throwing our own error when `window.ethereum` is not present allows to consistently capture the error.